tools/rna_tools/sortmerna/sortmerna.xml

```diff
@@ -220,6 +196,46 @@ sortmerna --version 2>&1|grep 'SortMeRNA version'
                 <expand macro="db_prep"/>
             </when>
         </conditional>
+        <conditional name="input_type">
+            <param name="input_type_selector" type="select" label="Input type">
+                <option value="not_paired">Reads are not paired</option><!-- single file -->
+                <option value="paired_one_file">Reads are paired (interleaved) in a single file</option>
+                <option value="paired_two_files">Reads are paired with Forward and Reverse strands in separate files</option>
+            </param>
+            <when value="not_paired">
+                <param argument="--reads" type="data" format="fasta,fastq" label="Querying sequences"/>
+            </when>
+            <when value="paired_one_file">
+                <param argument="--reads" type="data" format="fasta,fastq" label="Interleaved querying sequences"/>
+                <param name="strand_separation_type" type="select" display="checkboxes" optional="true" label="Output paired strands into separate files">
+                    <option value="--out2">Output FWD and REV strands into separate files i.e. 'aligned_fwd' and 'aligned_rev' (--out2)</option>
+                    <option value="--sout">Output paired aligned, and singleton aligned reads into separate files i.e. 'aligned_paired' and 'aligned_singleton' (--sout)</option>
+                </param>
+                <param name="paired_output_type" type="select" display="radio" optional="true" label="If one of the paired-end reads aligns and the other one does not">
+                    <option value="">Leave the reads split between aligned and rejected files</option>
+                    <option value="--paired_in">Output both reads to aligned file (--paired_in)</option>
+                    <option value="--paired_out">Output both reads to rejected file (--paired_out)</option>
+                </param>
```

Comment:
```suggestion
                <conditional name="output_options">
                    <param name="strand_separation_type" type="select" optional="true" label="Output paired strands into separate files">
                        <option value="--out2">--out2: Output FWD and REV strands into separate files i.e. 'aligned_fwd' and 'aligned_rev'</option>
                        <option value="--sout">--sout: Output paired aligned, and singleton aligned reads into separate files i.e. 'aligned_paired' and 'aligned_singleton'</option>
                        <option value="--out2 --sout">Use both --out2 and --sout</option>
                    </param>
                    <when value="--out2">
                        <param name="paired_output_type" type="select" display="radio" optional="true" label="If one of the paired-end reads aligns and the other one does not">
                            <option value="">Leave the reads split between aligned and rejected files</option>
                            <option value="--paired_in">Output both reads to aligned file (--paired_in)</option>
                            <option value="--paired_out">Output both reads to rejected file (--paired_out)</option>
                        </param>
                    </when>
                    <when value="--stout"/>
                    <when value="--out2 --stout"/>
                </conditional>
```

Reply:
I think that this is the best way for that.

Reply:
thanks! It helps me to get a grip on the logic.